internal/broker/broker.go

```diff
@@ -455,6 +455,8 @@ func (b *Broker) IsAuthenticated(sessionID, authenticationData string) (string,
 		return AuthCancelled, string(msg), ctx.Err()
 	}
 
+	log.Debugf(context.Background(), "Authentication result for session %s: %s", sessionID, access)
```

Comment:
What if the logging happens as a defer function that tells how we exited (something similar to what we do in pam.go authentication function)?